xl/settings.py

```diff
@@ -305,20 +305,19 @@ def save(self):
         logger.debug("Saving settings...")
 
         with open(self.location + ".new", 'w') as f:
-            self.write(f)
-
             try:
-                # make it readable by current user only, to protect private data
-                os.fchmod(f.fileno(), 384)
+                # make it readable by current user only
+                os.fchmod(f, S_IRUSR | S_IWUSR)
```

Comment:
These should probably be ```stat.S_IRUSR``` and ```stat.S_IWUSR```, unless I missed an import somewhere. Otherwise, this will silently fail.

Reply:
Using ```stat.S_IRUSR``` will require ```import stat```.

Also, ```fchmod()``` requires file descriptor, so the first argument should be ```f.fileno()``` as in original version.

Reply:
Thanks! I've updated the code now but I haven't tested it yet so I'm marking it as [WIP].

Reply:
Looks (and works) OK now.